rmw_fastrtps_cpp/src/publisher.cpp

```diff
@@ -199,6 +199,13 @@ rmw_fastrtps_cpp::create_publisher(
   }
   info->type_support_ = fastdds_type;
 
+  if (!rmw_fastrtps_shared_cpp::register_type_object(type_supports, type_name)) {
```

Comment:
is this related to the PR?
could you explain how if it is?

Reply:
Ok, that sounds fine, I will have to take a deeper look at the code.

Reply:
Just FYI, @iuhilnehc-ynos took inspiration from the auto-generated code of the Fast DDS unit tests [here](https://github.com/eProsima/Fast-DDS/tree/master/test/unittest/dds/topic/DDSSQLFilter/data_types)

Reply:
> Just FYI, @iuhilnehc-ynos took inspiration from the auto-generated code of the Fast DDS unit tests [here](https://github.com/eProsima/Fast-DDS/tree/master/test/unittest/dds/topic/DDSSQLFilter/data_types)

Could you point out what part of the generated code?

Reply:
https://github.com/eProsima/Fast-DDS/blob/caa3fe0ff9dc6a0ec31b08b0e44ab45d4892d7c5/test/unittest/dds/topic/DDSSQLFilter/data_types/ContentFilterTestTypeTypeObject.cxx